### PR TITLE
[MDB IGNORE] Moving broken/burned icons onto the flooring decl.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
Allows flooring decls to validate or override broken/burned states on a per-icon basis. Needs testing.

### CODE REVIEW

**Comment by MistakeNot4892** [comment]
Tested via explosion and fire on Tradeship.

**Comment by out-of-phaze** [comment]
> Tested via explosion and fire on Tradeship.

was it tested with wood floor tiles? previously it only worked for tile

**Comment by MistakeNot4892** [comment]
It's only set up on plating and tiles. Wood etc don't have broken states defined currently and will need them added.

File: code/unit_tests/turf_icons.dm
Comment (out-of-phaze) [inline code/unit_tests/turf_icons.dm:1]: line 18, `if(istype(floor))` should be `if(istype(floor, floor_type))` and also `var/turf/floor/check_floor` might just need to be `var/turf/check_turf` tbh... actually that var is unnecessary it should just do `if(TYPE_IS_ABSTRACT(floor_type))` shouldn't it

**Review by out-of-phaze — CHANGES_REQUESTED** [review]

**Review by out-of-phaze — DISMISSED** [review]

**Review by out-of-phaze — APPROVED** [review]

**Pull request merged**